providers/mimetypes.rb

```diff
@@ -0,0 +1,68 @@
+#
+## Cookbook Name:: gecos-ws-mgmt
+## Provider:: pdf_viewer
+##
+## Copyright 2013, Junta de Andalucia
+## http://www.juntadeandalucia.es/
+##
+## All rights reserved - EUPL License V 1.1
+## http://www.osor.eu/eupl
+##
+
+require 'etc'
+require 'chef/mixin/shell_out'
+include Chef::Mixin::ShellOut
+
+action :setup do
+	begin
+	  if new_resource.support_os.include?($gecos_os)
+    	package "xdg-utils" do
+       action :nothing
+      end.run_action(:install)
+
+      users = new_resource.users
+
+      users.each_key do |user_key|
+        nameuser = user_key
+        username = nameuser.gsub('###','.')
+        user = users[user_key]
+				gid = Etc.getpwnam(username).gid
+				xdg_code = ''
+
+				directory "/home/#{username}/.local/share/applications" do
+					owner username
+					group gid
+					recursive true
+				end
+
+				Chef::Log.debug("mimetypes.rb - Users: #{user}")
+
+				user.mimetyperelationship.each do |assoc|
```

Comment:
Buenas, Alfonso:

Al final, siguiendo tus indicaciones, parseo el fichero
$HOME/.local/share/applications/mimeapps.list del usuario y compruebo si
hay alguna modificación. Es mucho menos invasivo que lanzar un comando
shell por cada tipo mime configurado en la receta. Subo el código.

Saludos.

2016-08-03 14:58 GMT+02:00 Alfonso E.M. <notifications@github.com>:

> In providers/mimetypes.rb
> <https://github.com/gecos-team/gecos-workstation-management-cookbook/pull/145#discussion_r73332786>
> :
>
> > +      users.each_key do |user_key|
> > +        nameuser = user_key
> > +        username = nameuser.gsub('###','.')
> > +        user = users[user_key]
> > +				gid = Etc.getpwnam(username).gid
> > +				xdg_code = ''
> > +
> > +				directory "/home/#{username}/.local/share/applications" do
> > +					owner username
> > +					group gid
> > +					recursive true
> > +				end
> > +
> > +				Chef::Log.debug("mimetypes.rb - Users: #{user}")
> > +
> > +				user.mimetyperelationship.each do |assoc|
>
> Si estás cambiando el tipo mime para un usuario, puedes consultar las
> asociaciones actuales en $HOME/.local/mimeapps.list
>
> Para el sistema entero, están en /usr/share/applications/defaults.list
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/gecos-team/gecos-workstation-management-cookbook/pull/145/files/c25d460433d1f0d07780a75a6172e8a068ae51dd#r73332786>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATHDTCiN5IvNuUAvUe-2M6RpxZzVFf4sks5qcJBYgaJpZM4JRrKY>
> .
>